packages/react-api/src/Api.tsx

```diff
@@ -186,40 +188,76 @@ function Api ({ children, store, url }: Props): React.ReactElement<Props> | null
   const [extensions, setExtensions] = useState<InjectedExtension[] | undefined>();
 
   const value = useMemo<ApiProps>(
-    () => ({ ...state, api, apiError, apiUrl: url, extensions, isApiConnected, isApiInitialized, isWaitingInjected: !extensions }),
-    [apiError, extensions, isApiConnected, isApiInitialized, state, url]
+    () => ({ ...state, api, apiError, apiUrl: apiType?.param, extensions, isApiConnected, isApiInitialized, isWaitingInjected: !extensions }),
+    [apiError, extensions, isApiConnected, isApiInitialized, state, apiType?.param]
   );
 
   // initial initialization
   useEffect((): void => {
-    const provider = new WsProvider(url);
-    const signer = new ApiSigner(registry, queuePayload, queueSetTxStatus);
-    const types = getDevTypes();
-
-    api = new ApiPromise({ provider, registry, signer, types, typesBundle, typesChain });
-
-    api.on('connected', () => setIsApiConnected(true));
-    api.on('disconnected', () => setIsApiConnected(false));
-    api.on('error', (error: Error) => setApiError(error.message));
-    api.on('ready', (): void => {
-      const injectedPromise = web3Enable('polkadot-js/apps');
-
-      injectedPromise
-        .then(setExtensions)
-        .catch(console.error);
-
-      loadOnReady(api, injectedPromise, store, types)
-        .then(setState)
-        .catch((error): void => {
-          console.error(error);
-
-          setApiError((error as Error).message);
-        });
-    });
-
-    setIsApiInitialized(true);
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    async function substrateConnectApi (network: string) {
+      const detect = new Detector('PolkadotJS apps');
+      const signer = new ApiSigner(registry, queuePayload, queueSetTxStatus);
+      const types = getDevTypes();
+      const options = { registry, signer, types, typesBundle, typesChain };
+
+      try {
+        api = await detect.connect(network, undefined, options);
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        api.isReady.then(() => {
+          setIsApiInitialized(true);
+          setIsApiConnected(true);
+
+          const injectedPromise = web3Enable('polkadot-js/apps');
+
+          injectedPromise
+            .then(setExtensions)
+            .catch(console.error);
+
+          loadOnReady(api, injectedPromise, store, types)
+            .then(setState)
+            .catch((error): void => {
+              console.error(error);
+
+              setApiError((error as Error).message);
+            });
+        }).catch(console.error);
+      } catch (err) {
+        console.error(err);
+      }
+    }
+
+    if (apiType?.type === 'substrate-connect' as EndpointType) {
+      // eslint-disable-next-line no-void
+      void substrateConnectApi(apiType.param);
```

Comment:
```suggestion
     substrateConnectApi(apiType.param).catch(console.error);
```

Hate not handling the promises... just logging is 100% fine, but at least can see when something goes belly up.